### PR TITLE
Multiple improvements based on review comments

### PR DESCRIPTION
**Pull request by lionel1704**
- Use imports instead of mentioning the package in-line
- Use `PublicKey::from` instead of explicitly mentioning `PublicKey::Bls`
- Impl `Into` trait to convert `AppKeys` into `AppFullId`
- Re-use BLS key from `mock_routing::FullId`
- Combine `req_as_app` and `req_as_owner` functions for tests
- Use common signature verification for all requests

### CODE REVIEW

File: safe_core/src/client/mock/routing.rs [hunk 1]
```diff
@@ -120,7 +132,10 @@ impl Routing {
             proxy_node_name: new_rand::random(),
         };
 
-        let bls_sk = BlsSecretKey::random();
+        let bls_sk = id
+            .as_ref()
```
Comment (m-cat) [inline safe_core/src/client/mock/routing.rs]: Is `as_ref` needed here? Looks like you're just creating a clone of the internal key.

File: safe_core/src/client/mock/vault.rs [hunk 2]
```diff
@@ -794,7 +750,7 @@ impl Vault {
                     });
                 Response::MutateUnseqMDataEntries(result)
             }
-            _ => panic!("RPC not handled"),
+            _other => panic!("{:?} RPC not handled", _other),
```
Comment (m-cat) [inline safe_core/src/client/mock/vault.rs]: Variables should only start with an underscore if they are not used, surprised clippy didn't complain about this.

File: safe_core/src/client/mock/routing.rs (hunk 1 above)
Comment (lionel1704) [reply safe_core/src/client/mock/routing.rs]: I needed to use as_ref here as id is an `Option` and `unwrap_or_else` causes a `move`.

File: safe_core/src/client/mock/vault.rs (hunk 2 above)
Comment (lionel1704) [reply safe_core/src/client/mock/vault.rs]: I just googled this and learnt that I can use any identifier as a wildcard :smile: 
I'll update this to `other`, thanks :+1:

**Comment by nbaksalyar** [comment]
This PR addressed review comments on https://github.com/maidsafe/safe_client_libs/pull/862#pullrequestreview-252834306

**Pull request merged**